File: NGLPCMilerAssemblies/NGLTrimbleServices/GlobalSettings.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Diagnostics;
using System.Globalization;
using System.IO;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Security;

namespace NGLTrimbleServices
{
  
    static class GlobalSettings
    {
        public static short gServerID = 0;
        public static string gLastError = "";
        public static bool gProcessRunning = false;

               
        public static void AddBaddAddressToArray(int BookControl, int LaneControl, clsAddress objOrig, clsAddress objDest, clsAddress objPCMOrig, clsAddress objPCMDest, string Message, double BatchID, ref clsPCMBadAddress[] arrBaddAddresses)
        {
            clsPCMBadAddress oBadAddress = new clsPCMBadAddress();
            {
                var withBlock = oBadAddress;
                withBlock.BookControl = BookControl;
                withBlock.LaneControl = LaneControl;
                withBlock.objOrig = objOrig;
                withBlock.objDest = objDest;
                withBlock.objPCMOrig = objPCMOrig;
                withBlock.objPCMDest = objPCMDest;
                withBlock.Message = Message;
                withBlock.BatchID = BatchID;
            }
            if (arrBaddAddresses == null)
            {
                var oldArrBaddAddresses = arrBaddAddresses;
                arrBaddAddresses = new clsPCMBadAddress[1];
                if (oldArrBaddAddresses != null)
                    Array.Copy(oldArrBaddAddresses, arrBaddAddresses, Math.Min(1, oldArrBaddAddresses.Length));
            }
            else
            {
                var oldArrBaddAddresses = arrBaddAddresses;
                arrBaddAddresses = new clsPCMBadAddress[arrBaddAddresses.Length + 1];
                if (oldArrBaddAddresses != null)
                    Array.Copy(oldArrBaddAddresses, arrBaddAddresses, Math.Min(arrBaddAddresses.Length + 1, oldArrBaddAddresses.Length));
            }
            arrBaddAddresses[arrBaddAddresses.Length - 1] = oBadAddress;
        }

        public static void AddStopToArray(short shtStopNumber, string strStopName, string strID1, string strID2, string strTruckName, int intTruckNumber, short shtSeqNbr, double dblDistToPrev, double dblTotalRouteCost, string strConsNumber, ref clsAllStop[] arrAllStops)
        {
            // create a new object
            clsAllStop oStop = new clsAllStop();
            {
                var withBlock = oStop;
                withBlock.StopNumber = shtStopNumber;
                withBlock.Stopname = strStopName;
                withBlock.ID1 = strID1;
                withBlock.ID2 = strID2;
                withBlock.TruckDesignator = strTruckName;
                withBlock.TruckNumber = intTruckNumber;
                withBlock.SeqNbr = shtSeqNbr;
                withBlock.DistToPrev = dblDistToPrev;
                withBlock.TotalRouteCost = dblTotalRouteCost;
                withBlock.ConsNumber = strConsNumber;
            }
            if (arrAllStops == null)
            {
                var oldArrAllStops = arrAllStops;
                arrAllStops = new clsAllStop[1];
                if (oldArrAllStops != null)
                    Array.Copy(oldArrAllStops, arrAllStops, Math.Min(1, oldArrAllStops.Length));
            }
            else
            {
                var oldArrAllStops = arrAllStops;
                arrAllStops = new clsAllStop[arrAllStops.Length + 1];
                if (oldArrAllStops != null)
                    Array.Copy(oldArrAllStops, arrAllStops, Math.Min(arrAllStops.Length + 1, oldArrAllStops.Length));
            }
            arrAllStops[arrAllStops.Length - 1] = oStop;
        }

        public static void AddPCMDataStopToArray(int BookControl, int BookCustCompControl, int BookLoadControl, int BookODControl, int BookStopNo, int RouteType, int DistType, string BookOrigZip, string BookDestZip, string BookOrigAddress1, string BookDestAddress1, string BookOrigCity, string BookDestCity, string BookOrigState, string BookDestState, string BookProNumber, bool LaneOriginAddressUse, ref clsPCMDataStop[] arrPCMDataStops)
        {
            // create a new object
            clsPCMDataStop oPCMDataStop = new clsPCMDataStop();
            {
                var withBlock = oPCMDataStop;
                withBlock.BookControl = BookControl;
                withBlock.BookCustCompControl = BookCustCompControl;
                withBlock.BookLoadControl = BookLoadControl;
                withBlock.BookODControl = BookODControl;
                withBlock.BookStopNo = BookStopNo;
                withBlock.RouteType = RouteType;
                withBlock.DistType = DistType;
                withBlock.BookOrigZip = BookOrigZip;
                withBlock.BookDestZip = BookDestZip;
                withBlock.BookOrigAddress1 = BookOrigAddress1;
                withBlock.BookDestAddress1 = BookDestAddress1;
                withBlock.BookOrigCity = BookOrigCity;
                withBlock.BookDestCity = BookDestCity;
                withBlock.BookOrigState = BookOrigState;
                withBlock.BookDestState = BookDestState;
                withBlock.BookProNumber = BookProNumber;
                withBlock.LaneOriginAddressUse = LaneOriginAddressUse;
            }
            if (arrPCMDataStops == null)
            {
                var oldArrPCMDataStops = arrPCMDataStops;
                arrPCMDataStops = new clsPCMDataStop[1];
                if (oldArrPCMDataStops != null)
                    Array.Copy(oldArrPCMDataStops, arrPCMDataStops, Math.Min(1, oldArrPCMDataStops.Length));
            }
            else
            {
                var oldArrPCMDataStops = arrPCMDataStops;
                arrPCMDataStops = new clsPCMDataStop[arrPCMDataStops.Length + 1];
                if (oldArrPCMDataStops != null)
                    Array.Copy(oldArrPCMDataStops, arrPCMDataStops, Math.Min(arrPCMDataStops.Length + 1, oldArrPCMDataStops.Length));
            }
            arrPCMDataStops[arrPCMDataStops.Length - 1] = oPCMDataStop;
        }




}
}

File: NGLPCMilerAssemblies/NGLPCMilerTestApp/GlobalSettings.vb


Module GlobalSettings
    Public gServerID As Short = 0
    Public gLastError As String = ""
    Public gProcessRunning As Boolean = False

    Public Sub AddBaddAddressToArray(ByVal BookControl As Integer, _
                            ByVal LaneControl As Integer, _
                            ByVal objOrig As clsAddress, _
                            ByVal objDest As clsAddress, _
                            ByVal objPCMOrig As clsAddress, _
                            ByVal objPCMDest As clsAddress, _
                            ByVal Message As String, _
                            ByVal BatchID As Double, _
                            ByRef arrBaddAddresses() As clsPCMBadAddress)




        Dim oBadAddress As New clsPCMBadAddress
        With oBadAddress
            .BookControl = BookControl
            .LaneControl = LaneControl
            .objOrig = objOrig
            .objDest = objDest
            .objPCMOrig = objPCMOrig
            .objPCMDest = objPCMDest
            .Message = Message
            .BatchID = BatchID
        End With
        If arrBaddAddresses Is Nothing Then
            ReDim Preserve arrBaddAddresses(0)
        Else
            ReDim Preserve arrBaddAddresses(arrBaddAddresses.Length)
        End If
        arrBaddAddresses(arrBaddAddresses.Length - 1) = oBadAddress
    End Sub

    Public Sub AddStopToArray(ByVal shtStopNumber As Short, _
                    ByVal strStopName As String, _
                    ByVal strID1 As String, _
                    ByVal strID2 As String, _
                    ByVal strTruckName As String, _
                    ByVal intTruckNumber As Integer, _
                    ByVal shtSeqNbr As Short, _
                    ByVal dblDistToPrev As Double, _
                    ByVal dblTotalRouteCost As Double, _
                    ByVal strConsNumber As String, _
                    ByRef arrAllStops() As clsAllStop)
        'create a new object
        Dim oStop As New clsAllStop
        With oStop
            .StopNumber = shtStopNumber
            .Stopname = strStopName
            .ID1 = strID1
            .ID2 = strID2
            .TruckDesignator = strTruckName
            .TruckNumber = intTruckNumber
            .SeqNbr = shtSeqNbr
            .DistToPrev = dblDistToPrev
            .TotalRouteCost = dblTotalRouteCost
            .ConsNumber = strConsNumber
        End With
        If arrAllStops Is Nothing Then
            ReDim Preserve arrAllStops(0)
        Else
            ReDim Preserve arrAllStops(arrAllStops.Length)
        End If
        arrAllStops(arrAllStops.Length - 1) = oStop
    End Sub

    Public Sub AddPCMDataStopToArray(ByVal BookControl As Integer, _
                                ByVal BookCustCompControl As Integer, _
                                ByVal BookLoadControl As Integer, _
                                ByVal BookODControl As Integer, _
                                ByVal BookStopNo As Integer, _
                                ByVal RouteType As Integer, _
                                ByVal DistType As Integer, _
                                ByVal BookOrigZip As String, _
                                ByVal BookDestZip As String, _
                                ByVal BookOrigAddress1 As String, _
                                ByVal BookDestAddress1 As String, _
                                ByVal BookOrigCity As String, _
                                ByVal BookDestCity As String, _
                                ByVal BookOrigState As String, _
                                ByVal BookDestState As String, _
                                ByVal BookProNumber As String, _
                                ByVal LaneOriginAddressUse As Boolean, _
                                ByRef arrPCMDataStops() As clsPCMDataStop)
        'create a new object
        Dim oPCMDataStop As New clsPCMDataStop
        With oPCMDataStop
            .BookControl = BookControl
            .BookCustCompControl = BookCustCompControl
            .BookLoadControl = BookLoadControl
            .BookODControl = BookODControl
            .BookStopNo = BookStopNo
            .RouteType = RouteType
            .DistType = DistType
            .BookOrigZip = BookOrigZip
            .BookDestZip = BookDestZip
            .BookOrigAddress1 = BookOrigAddress1
            .BookDestAddress1 = BookDestAddress1
            .BookOrigCity = BookOrigCity
            .BookDestCity = BookDestCity
            .BookOrigState = BookOrigState
            .BookDestState = BookDestState
            .BookProNumber = BookProNumber
            .LaneOriginAddressUse = LaneOriginAddressUse
        End With
        If arrPCMDataStops Is Nothing Then
            ReDim Preserve arrPCMDataStops(0)
        Else
            ReDim Preserve arrPCMDataStops(arrPCMDataStops.Length)
        End If
        arrPCMDataStops(arrPCMDataStops.Length - 1) = oPCMDataStop

    End Sub

End Module
